### ISSUE TITLE
WARN [SocketServer brokerId=1] Unexpected error from /IPaddress; closing connection (org.apache.kafka.common.network.Selector)

### ISSUE THREAD

**Issue by max70-root** [strimzi/strimzi-kafka-operator#3462]
Hi Team,
i have a 3 node kafka cluster running on kubernetes with quite a standard configuration (see below) and external node ports exposed. The cluster works as expected, the only problem i have is a bunch of log messages in broker nodes

```
apiVersion: kafka.strimzi.io/v1beta1
kind: Kafka
metadata:
  name: my-cluster
spec:
  kafka:
    version: 2.5.0
    replicas: 3
    listeners:
      plain: {}
      tls: {}
      external:
        type: nodeport
        tls: false
    config:
      offsets.topic.replication.factor: 3
      transaction.state.log.replication.factor: 3
      transaction.state.log.min.isr: 2
      log.message.format.version: "2.5"
    storage:
      type: jbod
      volumes:
      - id: 0
        type: persistent-claim
        size: 30Gi
        class: manual
        deleteClaim: false
  zookeeper:
    replicas: 3
    storage:
      type: persistent-claim
      size: 30Gi
      class: manual
      deleteClaim: false
  entityOperator:
    topicOperator: {}
    userOperator: {}

```

Here's a sample of the messages i get:

```
2020-08-05 15:32:44,650 WARN [SocketServer brokerId=1] Unexpected error from /IP_ADDRESS; closing connection (org.apache.kafka.common.network.Selector) [data-plane-kafka-network-thread-1-ListenerName(EXTERNAL-9094)-PLAINTEXT-9]
org.apache.kafka.common.network.InvalidReceiveException: Invalid receive (size = 369296128 larger than 104857600)
        at org.apache.kafka.common.network.NetworkReceive.readFrom(NetworkReceive.java:105)
        at org.apache.kafka.common.network.KafkaChannel.receive(KafkaChannel.java:448)
        at org.apache.kafka.common.network.KafkaChannel.read(KafkaChannel.java:398)
        at org.apache.kafka.common.network.Selector.attemptRead(Selector.java:678)
        at org.apache.kafka.common.network.Selector.pollSelectionKeys(Selector.java:580)
        at org.apache.kafka.common.network.Selector.poll(Selector.java:485)
        at kafka.network.Processor.poll(SocketServer.scala:861)
        at kafka.network.Processor.run(SocketServer.scala:760)
        at java.base/java.lang.Thread.run(Thread.java:834)
2020-08-05 15:32:44,650 WARN [SocketServer brokerId=1] Unexpected error from /IP_ADDRESS; closing connection (org.apache.kafka.common.network.Selector) [data-plane-kafka-network-thread-1-ListenerName(EXTERNAL-9094)-PLAINTEXT-10]
org.apache.kafka.common.network.InvalidReceiveException: Invalid receive (size = 369296128 larger than 104857600)
        at org.apache.kafka.common.network.NetworkReceive.readFrom(NetworkReceive.java:105)
        at org.apache.kafka.common.network.KafkaChannel.receive(KafkaChannel.java:448)
        at org.apache.kafka.common.network.KafkaChannel.read(KafkaChannel.java:398)
        at org.apache.kafka.common.network.Selector.attemptRead(Selector.java:678)
        at org.apache.kafka.common.network.Selector.pollSelectionKeys(Selector.java:580)
        at org.apache.kafka.common.network.Selector.poll(Selector.java:485)
        at kafka.network.Processor.poll(SocketServer.scala:861)
        at kafka.network.Processor.run(SocketServer.scala:760)
        at java.base/java.lang.Thread.run(Thread.java:834)
2020-08-05 15:32:44,727 WARN [SocketServer brokerId=1] Unexpected error from /IP_ADDRESS; closing connection (org.apache.kafka.common.network.Selector) [data-plane-kafka-network-thread-1-ListenerName(EXTERNAL-9094)-PLAINTEXT-11]
org.apache.kafka.common.network.InvalidReceiveException: Invalid receive (size = 369296128 larger than 104857600)
        at org.apache.kafka.common.network.NetworkReceive.readFrom(NetworkReceive.java:105)
        at org.apache.kafka.common.network.KafkaChannel.receive(KafkaChannel.java:448)
        at org.apache.kafka.common.network.KafkaChannel.read(KafkaChannel.java:398)
        at org.apache.kafka.common.network.Selector.attemptRead(Selector.java:678)
        at org.apache.kafka.common.network.Selector.pollSelectionKeys(Selector.java:580)
        at org.apache.kafka.common.network.Selector.poll(Selector.java:485)
        at kafka.network.Processor.poll(SocketServer.scala:861)
        at kafka.network.Processor.run(SocketServer.scala:760)
        at java.base/java.lang.Thread.run(Thread.java:834)
2020-08-05 15:32:45,412 WARN [SocketServer brokerId=1] Unexpected error from /IP_ADDRESS; closing connection (org.apache.kafka.common.network.Selector) [data-plane-kafka-network-thread-1-ListenerName(EXTERNAL-9094)-PLAINTEXT-9]
org.apache.kafka.common.network.InvalidReceiveException: Invalid receive (size = 369296128 larger than 104857600)
        at org.apache.kafka.common.network.NetworkReceive.readFrom(NetworkReceive.java:105)
        at org.apache.kafka.common.network.KafkaChannel.receive(KafkaChannel.java:448)
        at org.apache.kafka.common.network.KafkaChannel.read(KafkaChannel.java:398)
        at org.apache.kafka.common.network.Selector.attemptRead(Selector.java:678)
        at org.apache.kafka.common.network.Selector.pollSelectionKeys(Selector.java:580)
        at org.apache.kafka.common.network.Selector.poll(Selector.java:485)
        at kafka.network.Processor.poll(SocketServer.scala:861)
        at kafka.network.Processor.run(SocketServer.scala:760)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

Would you please help me figuring out what's going on? I thing it is somehow related to external nodeport configuration or ssl but i really don't understand what i'm doing wrong

thank you in advance

**Comment by scholzj** [comment]
My best guess is that these are some TLS clients connecting to the unencrypted node port interface. It could be also some other form non-kafka client I guess. But TLS versus plaintext is the usual situation where I saw this kind of error.

**Comment by max70-root** [comment]
Thank you for your speedy answer!
And you were quite right as usual. After some hours of traffic sniffing the storm ended by itself. Now the cluster is performing as expected and no more errors in log. May be some non kafka client was hitting nodeports.

Thank you again

**Comment by GabrielLiz** [comment]
hi @max70-root i'm facing exacly the same issue could you explain a little more how to fix it.

thank you.

**Comment by LinTechSo** [comment]
Hi, any updates ?

**Comment by Shl007** [comment]
Hello, any updates ?

**Comment by rybick** [comment]
I found this issue looking for solution for the same problem and as @scholzj mentioned, it was caused by encryption. My app was sending encrypted traffic to Kafka that didn't expect it to be encrypted.
Somewhere on the way, my client lib was setting "security.protocol", "sasl.jaas.config" and "sasl.mechanism" properties. I removed that and now it works.
So If you're having the same problem make sure your "security.protocol" is not set or set to `PLAINTEXT`
